pandas/core/format.py

```diff
@@ -1679,11 +1679,13 @@ class ExcelFormatter(object):
     def __init__(self, df, na_rep='', float_format=None, cols=None,
                  header=True, index=True, index_label=None, merge_cells=False,
                  inf_rep='inf'):
-        self.df = df
         self.rowcounter = 0
         self.na_rep = na_rep
-        self.columns = cols
-        if cols is None:
+        if cols is not None:
+            self.df = df.loc[:, cols]
+            self.columns = cols
+        else:
```

Comment:
if ``cols`` names a non-existant column ``.reindex`` will work, ``.iloc`` will raise. pls add a test for this (IIRC I think this is the same way ``.to_csv`` works, pls check that as they should be consistent)